auth/src/api/services/user-service.ts

```diff
@@ -63,17 +64,17 @@ export class UserService implements UserHandler {
         const userObj = await this.getUserCollectionFromDatastore('username', username);
         Logger.debug('User object from datastore: ', userObj);
         if (Object.keys(userObj).length > 1) {
-            throw new Error('Multiple users with same credentials!');
+            throw new AuthenticationException('Multiple users with same credentials!');
         }
         const user: User = new User(userObj[Object.keys(userObj)[0]]);
         if (!user.isExisting()) {
-            throw new Error('Username or password is incorrect!');
+            throw new AuthenticationException('Username or password is incorrect.');
         }
         if (!user.is_active) {
-            throw new Error('The account is deactivated.');
+            throw new AuthenticationException('The account is deactivated.');
         }
         if (!this.isPasswordCorrect(password, user.password)) {
```

Comment:
I'm just curious: Do you have tests for `password=''` and `password=null`?

Reply:
Why do you think, this is necessary?
I think, a submitted password is compared to a stored one. If the submitted password is `'' || null || undefined`, this won't be equal to a stored password.

Reply:
Just, that the service can handle null-inputs and do not raise strange errors (like trying to hash it or so).